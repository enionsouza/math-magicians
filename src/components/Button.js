import PropTypes from 'prop-types';

const Button = ({
  action, label, cssClass, cssStyle,
}) => (
  <button type="button" style={cssStyle} onClick={action} className={cssClass}>{label}</button>
);

Button.propTypes = {
  action: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  cssClass: PropTypes.string.isRequired,
  cssStyle: PropTypes.shape({
    gridArea: PropTypes.string,
  }).isRequired,
};

export default Button;
