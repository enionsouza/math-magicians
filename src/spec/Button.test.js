import { render, fireEvent, screen } from '@testing-library/react';
import Button from '../components/Button';

describe('Button component', () => {
  describe('Use Jest snapshots to test the Button component', () => {
    it('renders a button with the correct label', () => {
      const actionHandlerMock = jest.fn(() => {});
      const { asFragment } = render(<Button action={actionHandlerMock} label="Button Label" cssClass="default-class" cssStyle={{ gridArea: 'grid-1' }} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('Simulate user interaction with Button component', () => {
    it('calls the action prop function when user clicks on it', () => {
      const actionHandlerMock = jest.fn(() => {});
      render(<Button action={actionHandlerMock} label="Button Label" cssClass="default-class" cssStyle={{ gridArea: 'grid-1' }} />);
      fireEvent.click(screen.getByText('Button Label'));
      expect(actionHandlerMock.mock.calls.length).toBe(1);
    });
  });
});
