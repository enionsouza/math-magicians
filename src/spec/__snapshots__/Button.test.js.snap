// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Use Jest snapshots to test the Button component renders a button with the correct label 1`] = `
<DocumentFragment>
  <button
    class="default-class"
    style="grid-area: grid-1;"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;
